lib/ansible/modules/cloud/podman/podman_container.py

```diff
@@ -0,0 +1,1213 @@
+        if params.get('stop_timeout'):
+            cmd += ['--stop-timeout', str(params['stop_timeout'])]
+        if params.get('subgidname'):
+            cmd += ['--subgidname', params['subgidname']]
+        if params.get('subuidname'):
+            cmd += ['--subuidname', params['subuidname']]
+        if params.get('sysctl'):
+            cmd += ['--sysctl', params['sysctl']]
+        if params.get('systemd') is not None:
+            cmd += ['--systemd=' + str(params['systemd'])]
+        if params.get('tmpfs'):
+            cmd += ['--tmpfs', params['tmpfs']]
+        if params.get('tty') is not None:
+            cmd += ['--tty=' + str(params['tty'])]
+        if params.get('uidmap'):
+            cmd += ['--uidmap', params['uidmap']]
+        if params.get('ulimit'):
+            cmd += ['--ulimit', params['ulimit']]
+        if params.get('user'):
+            cmd += ['--user', params['user']]
+        if params.get('userns'):
+            cmd += ['--userns', params['userns']]
+        if params.get('uts'):
+            cmd += ['--uts', params['uts']]
+        if params.get('volume'):
+            for vol in params['volume']:
+                cmd += ['--volume', vol]
+        if params.get('volumes_from'):
+            for vol in params['volumes_from']:
+                cmd += ['--volumes-from', vol]
+        if params.get('workdir'):
+            cmd += ['--workdir', params['workdir']]
+        # Add your own args for podman command
+        if params.get('cmd_args'):
+            cmd += params['cmd_args']
+        cmd.append(params['image'])
+        if params.get('command'):
+            cmd.append(params['command'])
+        return cmd
+
+
+def ensure_image_exists(module, image):
+    rc, out, err = run_podman_command(module, args=['image', 'exists', image], ignore_errors=True)
+    if rc != 0:
+        rc, out, err = run_podman_command(module, args=['image', 'pull', image])
+        if rc != 0:
+            module.fail_json(msg="Can't pull image %s:\n%s\v%s" % (image, out, err))
+
+
+class PodmanContainer(object):
+    def __init__(self, module, name, params):
+
+        super(PodmanContainer, self).__init__()
+        self.module = module
+        self.name = name
+        self.info = self.get_info()
+
+    @property
+    def exists(self):
+        return self.info != {}
+
+    @property
+    def different(self):
+        # TODO(sshnaidm): implement difference calculation between input vars
+        # and current container to understand if we need to recreate it
+        return True
+
+    @property
+    def running(self):
+        return False if not self.exists else self.info['State']['Running']
+
+    @property
+    def stopped(self):
+        return False if not self.exists else not self.info['State']['Running']
+
+    def get_info(self):
+        rc, out, err = run_podman_command(
+            module=self.module,
+            args=['container', 'inspect', self.name], ignore_errors=True)
+        if rc != 0:
```

Comment:
I recommend you reversing this into a "guard expression" style:
```suggestion
        if rc == 0:
        	return json.loads(out)[0]
```
And then you can reduce the nesting of the whole block of code which improves readability.

Reply:
Went over a code to do it, thanks